cmd/cortex/Dockerfile

```diff
@@ -1,7 +1,22 @@
 FROM       alpine:3.13
-RUN        apk add --no-cache ca-certificates
+RUN        apk add --no-cache ca-certificates libcap
+
+# create cortex user to run cortex as non-root
+RUN addgroup -g 10000 -S cortex && \
+    adduser  -u 10000 -S cortex -G cortex
+
+RUN mkdir /data && \
+    chown cortex:cortex /data
+
+VOLUME /data
+WORKDIR /data
+
 COPY       migrations /migrations/
 COPY       cortex /bin/cortex
+
+# allow cortex to bind to port 80 as non-root
+RUN setcap 'cap_net_bind_service=+ep' /bin/cortex
```

Comment:
[question] Do you see any problem with this change when running the image in K8S?

Reply:
I think the way it should work is by specifying in the manifest:
```
      securityContext:
        capabilities:
          add:
            - NET_BIND_SERVICE
```

However it looks like this doesn't work: https://github.com/kubernetes/kubernetes/issues/56374

Reply:
Would it be an idea to add a `USER` line, so it no longer defaults to root?
I guess this might cause trouble upgrading an installation, which has existing files in persistent volumes.

Reply:
That's what I eventually would like to get to, but in a first step, I want users of the docker image to opt in to not run as root. Otherwise we will need to ensure mounted in persistent volumes are `chown`ed first.

> `securityContext: ...` 

I think if we rely on the container orchestration to set it up (k8s in that case), we will break users that are using bare docker. I think without that setcap we probably need to change the default port 80 to something > 1024